### PR TITLE
Fixes #16781 adds an 100Base SFP interface

### LINKED ISSUE #11371: Interface Type Addition: Add 100Base-FX and other 100Mbps Types
### NetBox version

v3.4.1

### Feature type

Data model extension

### Proposed functionality

We have some legacy hardware that runs on 100Base-FX (Fast Ethernet over Fiber). We would like to request the interface type be added

### Use case

We have some Allied Telesis AT-FS202's in the field that we are looking to record. These have 100Base-FX connection for their Port1 and 10Base-T/100Base-TX for their Port 2

### Database changes

_No response_

### External dependencies

_No response_

### LINKED ISSUE #16781: New interface type SFP (100 Base-FX)
### NetBox version

v4.0.6

### Feature type

Data model extension

### Proposed functionality

Add 100Mbps SFP interface type (100 Base FX)

### Use case

I'm documenting an active network device that we have: Alcatel 7250 SAS 
Specs: https://www.starmicrowave.com/wp-content/uploads/2018/03/7250SAS_v2.pdf

I tried to create 100Mbps SFP interface but netbox only has SFP (1GE). This device has both 100Mbps and 1GB. It would be useful to differentiate

### Database changes

_No response_

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by RobertH1993**
### Fixes: #16781

Adds the requested SFP type

### CODE REVIEW

**Comment by RobertH1993** [comment]
Comment so maintainer is able to assign.

File: netbox/dcim/choices.py [hunk 1]
```diff
@@ -825,6 +825,7 @@ class InterfaceTypeChoices(ChoiceSet):
     # Ethernet
     TYPE_100ME_FX = '100base-fx'
     TYPE_100ME_LFX = '100base-lfx'
+    TYPE_100ME_FX_SFP = '100base-fx-sfp'
```
Comment (jeremystretch) [inline netbox/dcim/choices.py:828]:
```suggestion
    TYPE_100ME_SFP = '100base-x-sfp'
```

File: netbox/dcim/choices.py [hunk 2]
```diff
@@ -1000,6 +1002,7 @@ class InterfaceTypeChoices(ChoiceSet):
             (
                 (TYPE_1GE_GBIC, 'GBIC (1GE)'),
                 (TYPE_1GE_SFP, 'SFP (1GE)'),
+
```
Comment (jeremystretch) [inline netbox/dcim/choices.py:1005]: Remove errant newline.
```suggestion
```

File: netbox/dcim/choices.py [hunk 3]
```diff
@@ -985,6 +986,7 @@ class InterfaceTypeChoices(ChoiceSet):
             (
                 (TYPE_100ME_FX, '100BASE-FX (10/100ME FIBER)'),
                 (TYPE_100ME_LFX, '100BASE-LFX (10/100ME FIBER)'),
+                (TYPE_100ME_FX_SFP, '100BASE-FX-SFP (10/100ME FIBER)'),
```
Comment (jeremystretch) [inline netbox/dcim/choices.py:989]: The SFP interface type does not imply a specific medium (e.g. FX), just the line rate. Also, this needs to be moved to the "Ethernet (modular)" group as it is modular in nature.
```suggestion
                (TYPE_100ME_SFP, 'SFP (100ME)'),
```

**Review by jeremystretch — CHANGES_REQUESTED** [review]

**Pull request closed without merging**